hips/hip-9997.md

```diff
@@ -0,0 +1,197 @@
+
+### 4. Chart names == OCI reference basenames
+
+Again, to keep things simple, the basename (the last segment of the URL path) on a registry reference should be equivalent to the chart name. 
+
+For example, given a chart with the name `pepper` and the version `1.2.3`, users may run a command such as the following:
+
+```
+$ helm push pepper/ oci://r.myreg.io/mycharts
+```
+
+which would result in the following reference:
+
+```
+oci://r.myreg.io/mycharts/pepper:1.2.3
+```
+
+By placing such restrictions on registry URLs and tags, Helm users are less likely to do "strange things" with charts in registries.
+
+### 5. Cache is removed
+
+Since chart packages are small in size (<1mb), the cache is hard to justify. The cache was introduced only to provide a "Docker-like" experience. While this is neat, it does not provide the user with much value.
+
+By removing the cache, much of the existing OCI features can be cut down dramatically in size. This will make OCI features much more seamless with the existing Helm user experience.
+
+### 6. `helm chart` is removed, integrated into rest of CLI
+
+Wherever possible, the subcommands provided by the new `helm chart` command should be integrated into existing Helm CLI commands.
+
+`helm chart pull` should be baked into `helm pull`, `helm chart push` should be `helm push` (new).
+
+Commands that work with the cache will be removed: `helm chart save`, `helm chart remove` and `helm chart export`.
+
+`helm registry` will be kept as is. This manages auth against OCI registries.
+
+
+### 7. Experimental until clear messaging from OCI
+
+Considering that the rest of the items above are implemented and resolved, the OCI feature set will not be made generally available until there is clear messaging from the Open Container Initiative (OCI) regarding the way that Helm is using registries.
+
+The technical details of how Helm is using the OCI distribution spec must be validated in writing, in a document approved by OCI, clearly stating that this is the correct way to publish an arbitrary artifact. Only after this will these features be taken out of experimental mode and made generally available.
+
+## Backwards compatibility
+
+Since the existing feature set is currently experimental, there will be no promise of backwards compatibility with prior OCI support (sorry!)
+
+## Security implications
+
+Registry authentication introduces a new attack vector. Upon running `helm registry login`, these credentials are stored in an unencrypted JSON file. With that said, the underlying code is leveraging docker/cli, which has the ability to store these credentials into the system keychain instead.
+
+So, while a new attack vector has been introduced, it should not be considered any less safe than using `docker login`.
```

Comment:
`helm registry login` actually does use credentials helpers just as `docker login`:

```
$ cat $(helm env | grep HELM_REGISTRY_CONFIG | cut -d= -f2 | sed 's/"//g')
{
	"auths": {
		"bundle.bar": {},
		"localhost:5001": {}
	},
	"credsStore": "osxkeychain"
}
```

Do you still disagree? Is there some language I can add to make this clearer?

Reply:
Oh! I wasn't aware that was possible. My apologies.

One line 142: "Upon running `helm registry login`, these credentials are stored in an unencrypted JSON file." That to me implies that keychain support did not exist.

If it does, can we remove that statement, or perhaps amend it to say that `helm registry login` has keychain support?

Reply:
I agree, The first paragraph here (line 142) needs cleaning up. It should _lead_ with the second sentence "we use `docker/cli` and so share credential storage configuration mechanisms with `docker login`", and follow up with "*WARNING* if credential storage is not configured, your credentials will end up on disk in an unprotected JSON file" or similar, per the first sentence.

Basically, all the information is there, but it's presented in a confusing order, and misplaces the emphasis.

It might be handy (if not already the case) if Helm could _read_ the Docker configuration too, so that a working Docker install (at least for Docker Desktop) is sufficient to secure Helm too, since it configures Windows Credential Store or OSX Keychain by default.

It would also save copying, e.g., ECR credential store configuration over from Docker config to Helm config.

Reply:
added some language to make this more clear